### PR TITLE
DEV-2

### PR DESCRIPTION
**Pull request by NikaMakovik**

### CODE REVIEW

File: DEV-2/DEV-2/EntryPoint.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+using System.Text;
+
+namespace DEV_2
+{
+    class EntryPoint
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Please enter your string: ");
+            StringBuilder theString = new StringBuilder(Console.ReadLine());
```
Comment (IgorKheidorov) [inline DEV-2/DEV-2/EntryPoint.cs]: Bad value name, better newStringCreator

Comment (IgorKheidorov) [inline DEV-2/DEV-2/EntryPoint.cs]: No need to create string builder here

File: DEV-2/DEV-2/StringLine.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+{
+    class StringCreator
+    {
+        /// <summary>
+        /// This Method checks possibility of using string.
+        /// </summary>
+        /// <param name="theString">The main string.</param>
+        private void CheckPossibilityOfUsing(StringBuilder theString)
+        {
+            if (theString.Length<1)
```
Comment (IgorKheidorov) [inline DEV-2/DEV-2/StringLine.cs]: code style

File: DEV-2/DEV-2/StringLine.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        /// <param name="theString">The main string.</param>
+        public void GetNewStringOddIndices(StringBuilder theString)
+        {
+            CheckPossibilityOfUsing(theString);
+            StringBuilder newString = new StringBuilder();
+            for (int i = 0; i < theString.Length; i+=2)
+            {
+                newString.Append(theString[i]);
+            }
+            Console.WriteLine("New string: ");
```
Comment (IgorKheidorov) [inline DEV-2/DEV-2/StringLine.cs]: Can unite to one line

File: DEV-2/DEV-2/StringLine.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            if (theString.Length<1)
+            {
+                Console.WriteLine("The string is empty. Try again. ");
+                return;
+            }
+        }
+        /// <summary>
+        /// This Method make a new string with odd indices from the main string.
+        /// </summary>
+        /// <param name="theString">The main string.</param>
```
Comment (IgorKheidorov) [inline DEV-2/DEV-2/StringLine.cs]: Two StringBuilders are not necessary

File: DEV-2/DEV-2/StringLine.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Text;
+
+namespace DEV_2
+{
+    class StringCreator
+    {
+        /// <summary>
```
Comment (IgorKheidorov) [inline DEV-2/DEV-2/StringLine.cs]: What do you mean about possibility of using. Every string can be used

**Review by IgorKheidorov — CHANGES_REQUESTED** [review]

File: DEV-2/DEV-2/EntryPoint.cs [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+{
+    class EntryPoint
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Please enter your string: ");
+            string theString = Console.ReadLine();
+            if (theString.Length < 1)
+            {
+                Console.WriteLine("The string is empty. Try again.");
```
Comment (AlexanderSoroka) [inline DEV-2/DEV-2/EntryPoint.cs]: Wrong message - you ask user to try again and just finish the application

File: DEV-2/DEV-2/EntryPoint.cs [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace DEV_2
+{
+    class EntryPoint
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Please enter your string: ");
+            string theString = Console.ReadLine();
+            if (theString.Length < 1)
```
Comment (AlexanderSoroka) [inline DEV-2/DEV-2/EntryPoint.cs]: that is not obvious. Either use == 0 either check for empty string

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]
Remove .gitignore/.gitignore → .gitignore from that changes

**Pull request merged**